TwitchDownloaderCLI/Modes/UpdateHandler.cs

```diff
@@ -0,0 +1,192 @@
+﻿using CommandLine.Text;
+using System;
+using System.Net.Http;
+using System.IO;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using System.Xml;
+using System.Linq;
+using System.IO.Compression;
+using TwitchDownloaderCLI.Modes.Arguments;
+using System.Reflection;
+using TwitchDownloaderCore.Extensions;
+using System.Runtime.InteropServices;
+
+namespace TwitchDownloaderCLI.Modes
+{
+    internal static class UpdateHandler
+    {
+        private static HttpClient _client = new();
+
+        public static void ParseArgs(UpdateArgs args)
+        {
+#if DEBUG
+            Console.WriteLine("Auto-update is not supported for debug builds");
+#else
+            CheckForUpdate(args.ForceUpdate).GetAwaiter().GetResult();
+#endif
+        }
+
+        private static async Task CheckForUpdate(bool forceUpdate)
+        {
+            // Get the old version
+            Version oldVersion = Assembly.GetExecutingAssembly().GetName().Version!.StripRevisionIfDefault();
+
+            // Get the new version
+            var updateInfoUrl = @"https://downloader-update.twitcharchives.workers.dev/";
+            using HttpResponseMessage response = await _client.GetAsync(updateInfoUrl);
+            string xmlString = await response.Content.ReadAsStringAsync();
+
+            if (string.IsNullOrEmpty(xmlString))
+            {
+                Console.Error.WriteLine("Internal error: could not parse remote update info XML!");
+            }
+
+            XmlDocument xmlDoc = new XmlDocument();
+            xmlDoc.LoadXml(xmlString);
+
+            string newVersionString = xmlDoc.DocumentElement.SelectSingleNode("/item/version").InnerText;
```

Comment:
Tell the compiler there won't be an NRE
```suggestion
            string newVersionString = xmlDoc.DocumentElement!.SelectSingleNode("/item/version").InnerText;
```

Reply:
Addressed in commit [1706996](https://github.com/lay295/TwitchDownloader/pull/1345/commits/1706996a5d6f1bfa25506e4cd273eeffb6487dfb)